pennylane/transforms/optimization/cancel_inverses.py

```diff
@@ -49,17 +39,39 @@ def _are_inverses(op1, op2):
     Returns:
         Bool
     """
-    # op1 is self-inverse and the next gate is also op1
-    if op1 in self_inverses and op1.name == op2.name:
-        return True
-
-    # op1 is an `Adjoint` class and its base is equal to op2
-    if isinstance(op1, Adjoint) and _ops_equal(op1.base, op2):
-        return True
-
-    # op2 is an `Adjoint` class and its base is equal to op1
-    if isinstance(op2, Adjoint) and _ops_equal(op2.base, op1):
-        return True
+    # Make sure that if one of the ops is Adjoint, it is always op2 by swapping
+    # the ops if op1 is Adjoint
+    if isinstance(op1, Adjoint):
+        op1, op2 = op2, op1
+
+    are_self_inverses_without_wires = op1 in self_inverses and op1.name == op2.name
+    are_inverses_without_wires = (
+        isinstance(op2, Adjoint)
+        and op1.__class__ == op2.base.__class__
+        and op1.data == op2.base.data
```

Comment:
How is this not raising errors? I'm actually a bit hesitant about directly comparing numeric data like this.

Reply:
We don't have a lot of interface testing for `cancel_inverses`, but using equality for floating point data is fine as long as any arithmetic ops aren't applied to the data. The only issue I see with the use of `==` is that we will not have any tolerance for comparison, leading to potentially cancellable ops not being cancelled.

Reply:
Maybe we can add a `tol` parameter to `cancel_inverses` for comparing numerical data. 🤔